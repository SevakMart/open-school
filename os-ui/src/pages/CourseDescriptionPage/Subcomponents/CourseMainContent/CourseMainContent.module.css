.mainContent {
  width: 50%;
  margin-top: 16px;
  align-self: flex-start;
}
.mainContent .mainTitle {
  display: inline;
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 800;
  font-size: 48px;
  line-height: 59px;
  color: #222427;
}
.mainContent .mainCourseInfoContainer {
  margin: 112px 0 56px;
}
.mainContent .mainCourseInfoContainer .subContent {
  margin-bottom: 32px;
}
.mainContent .mainCourseInfoContainer .subContent h2 {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 600;
  font-size: 20px;
  line-height: 24px;
  color: #333941;
  margin: 0 0 16px;
}
.mainContent .mainCourseInfoContainer .subContent p {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 400;
  font-size: 16px;
  line-height: 24px;
  color: #5E617B;
  text-align: justify;
  margin: 0;
}
.mainContent .mainCourseInfoContainer .subContent .moduleListContainer {
  border-radius: 20px;
  background-color: #ffffff;
  display: grid;
  grid-template-columns: repeat(1, 1fr);
  gap: 10px 0;
  justify-content: none;
  padding: 10px 0;
}/*# sourceMappingURL=CourseMainContent.module.css.map */