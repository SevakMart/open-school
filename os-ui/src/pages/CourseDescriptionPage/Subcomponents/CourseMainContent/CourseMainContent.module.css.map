{"version":3,"sources":["CourseMainContent.module.scss","CourseMainContent.module.css","../../../../styles/Mixins/FontMixin.scss","../../../../styles/Configs/Fonts.scss","../../../../styles/Configs/Colors.scss","../../../../styles/Mixins/GridMixin.scss"],"names":[],"mappings":"AAMA;EACI,UAAA;EACA,gBAAA;EACA,sBAAA;ACLJ;ADMG;EACC,eAAA;EEVF,yBCDY;EDEZ,kBFUmC;EETnC,gBFS0C;EER1C,eFQ8C;EEP9C,iBFOmD;EACjD,cIVgB;AHUpB;ADEG;EACC,oBAAA;ACAJ;ADCI;EACI,mBAAA;ACCR;ADAQ;EElBN,yBCDY;EDEZ,kBFkB2C;EEjB3C,gBFiBkD;EEhBlD,eFgBsD;EEftD,iBFe2D;EACjD,cIhBO;EJiBP,gBAAA;ACMZ;ADJQ;EEvBN,yBCDY;EDEZ,kBFuB2C;EEtB3C,gBFsBkD;EErBlD,eFqBsD;EEpBtD,iBFoB2D;EACjD,cIxBM;EJyBN,mBAAA;EACA,SAAA;ACUZ;ADRQ;EACI,mBAAA;EACA,yBI5BO;ECHf,aAAA;EACA,qCL+B2B;EK9B3B,WAAA;EACA,qBL6ByC;EACjC,eAAA;ACaZ","file":"CourseMainContent.module.css"}