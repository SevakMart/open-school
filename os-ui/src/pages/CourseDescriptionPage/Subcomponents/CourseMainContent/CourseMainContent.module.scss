@import "../../../../styles/Mixins/FontMixin.scss";
@import "../../../../styles/Mixins/FlexMixins.scss";

.mainContent{
    width:50%;
    margin-top:3rem;
    .subContent {
        h2 {
            @include fontStyle('Montserrat',normal,600, 1.25rem,1.5rem);
            color: #333941;
        }
        p{
            @include fontStyle('Montserrat',normal,400, 1rem,1.5rem);
            color: #5E617B;
            text-align: justify;
        }
        .moduleListContainer {
            @include flexStyle(column, center, space-between);
            padding-left:0.625rem;
            border:2px solid blue;
        }
    }
    .authorInformationContainer {
        h2 {
            @include fontStyle('Montserrat',normal,600, 1.25rem,1.5rem);
            color: #333941;
        }
        .authorInfo {
            @include flexStyle(row,center,space-between);
            width:30%;
            .imageAndFullName{
                @include flexStyle(row,center,space-between);
                width:60%;  
            img {
                border-radius: 50%;
                width:2rem;
                height:2rem;
                border-color: black;
                color:black;
            }
        }
           a {
            cursor:pointer;
           }
        }
    }
}

