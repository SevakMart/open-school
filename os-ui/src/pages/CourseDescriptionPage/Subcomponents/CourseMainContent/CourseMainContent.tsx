import { useTranslation } from 'react-i18next';
import { CourseDescriptionType } from '../../../../types/CourseTypes';
import ModuleItem from '../ModuleItem/ModuleItem';
import AuthorInfo from '../AuthorInfo/AuthorInfo';
import styles from './CourseMainContent.module.scss';
/* eslint-disable max-len */
const CourseMainContent = ({
  description, goal, modules, mentorDto, title, currentUserEnrolled,
}: Omit<CourseDescriptionType, 'Rating' | 'Enrolled' | 'Course Level' | 'Language' | 'Estimated efforts'>) => {
  const { t } = useTranslation();
  const {
    mainContent, subContent, moduleListContainer, mainTitle, mainCourseInfoContainer, titleContainer, enrolledTitleContainer,
    erolledTitle, erolledModuleListContainer,
  } = styles;

  return (
    <div className={mainContent}>
      <div className={currentUserEnrolled ? enrolledTitleContainer : titleContainer}>
        <h1 className={currentUserEnrolled ? erolledTitle : mainTitle}>
          {title}
        </h1>
      </div>
      <div className={mainCourseInfoContainer}>
        <div className={subContent}>
          <h2>{t('What You Will Learn')}</h2>
          <p>{description}</p>
        </div>
        <div className={subContent}>
          <h2>{t('Course Goal')}</h2>
          <p>{goal}</p>
        </div>
        <div className={subContent}>
          <h2>{t('Modules in This Program')}</h2>
          <div className={currentUserEnrolled ? erolledModuleListContainer : moduleListContainer}>
            {modules.map((module, index) => (<ModuleItem key={index} moduleInfo={module} />))}
          </div>
        </div>
      </div>
      <AuthorInfo mentorName={mentorDto.name} mentorSurname={mentorDto.surname} mentorLinkedIn={mentorDto.linkedinPath} />
    </div>
  );
};

export default CourseMainContent;
