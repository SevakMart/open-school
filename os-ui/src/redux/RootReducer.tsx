import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
