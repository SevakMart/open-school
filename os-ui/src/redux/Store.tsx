import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
