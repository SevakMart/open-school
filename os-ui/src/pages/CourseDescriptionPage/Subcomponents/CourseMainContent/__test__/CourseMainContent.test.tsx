import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../../../redux/Store';
import CourseMainContent from '../CourseMainContent';

const moduleInfo = {
  title: 'React js',
  description: 'It talks about React',
  moduleItemSet: [{ title: 'Redux' }],
};
const mockUseNavigate = jest.fn();
const mockUseLocation = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockUseNavigate,
  useLocation: () => mockUseLocation,
}));
jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key:string) => key }),
}));

describe('Create test cases for CourseMainContent', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(<Provider store={store}><CourseMainContent description="This course talks about react" goal="The goal is to gain experience." modules={[moduleInfo]} mentorDto={{ name: 'john', surname: 'Smith', linkedinPath: 'myLinkedin.com' }} title="React" currentUserEnrolled={false} /></Provider>, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });
});
