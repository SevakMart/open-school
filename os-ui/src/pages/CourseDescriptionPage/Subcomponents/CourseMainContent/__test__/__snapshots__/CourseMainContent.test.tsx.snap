// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for CourseMainContent Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mainContent"
  >
    <h1
      class="mainTitle"
    >
      React
    </h1>
    <div
      class="mainCourseInfoContainer"
    >
      <div
        class="subContent"
      >
        <h2>
          string.courseDescriptionPage.title.description
        </h2>
        <p>
          This course talks about react
        </p>
      </div>
      <div
        class="subContent"
      >
        <h2>
          string.courseDescriptionPage.title.goal
        </h2>
        <p>
          The goal is to gain experience.
        </p>
      </div>
      <div
        class="subContent"
      >
        <h2>
          string.courseDescriptionPage.title.modulesList
        </h2>
        <div
          class="moduleListContainer"
        >
          <div
            class="mainContainer"
          >
            <div
              class="mainContent"
            >
              <div
                class="titleAndEllipseIcon"
              >
                <div
                  class="titleAndIconWrapper"
                >
                  <img
                    alt="ellipse icon"
                    class="ellipseIcon"
                    src="Ellipse.svg"
                  />
                  <p
                    class="moduleTitle"
                  >
                    React js
                  </p>
                </div>
              </div>
              <img
                alt="chevron"
                class="chevronIsClosed"
                src="ArrowRight.svg"
              />
            </div>
            <div
              class="moduleDescriptionIsClosed"
            >
              <p
                class="moduleTitle"
              >
                It talks about React
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="authorInformationContainer"
    >
      <h2>
        string.courseDescriptionPage.title.authorInfo
      </h2>
      <div
        class="authorInfo"
      >
        <img
          alt="john Smith"
          class="mentorProfileImage"
          src="https://reactjs.org/logo-og.png"
        />
        <p>
          john Smith
        </p>
        <a
          href="myLinkedin.com"
        >
          <img
            alt="LinkedIn icon"
            src="LinkedIn.svg"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
